goldmane/fv/daemon_test.go

```diff
@@ -290,3 +290,72 @@ func TestHints(t *testing.T) {
 		return nil
 	}, 5*time.Second, 1*time.Second).Should(Succeed())
 }
+
+func TestStatistics(t *testing.T) {
+	cfg := daemon.Config{
+		LogLevel:                 "debug",
+		Port:                     8988,
+		AggregationWindow:        time.Second * 1,
+		EmitAfterSeconds:         2,
+		EmitterAggregationWindow: time.Second * 2,
+	}
+	defer daemonSetup(t, cfg)()
+
+	// Generate credentials for the Goldmane client.
+	creds, err := client.ClientCredentials(clientCert, clientKey, clientCA)
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to create goldmane TLS credentials.")
+	}
+
+	// Generate credentials for the Goldmane client.
+	statsClient, err := client.NewStatisticsAPIClient(goldmaneURL, grpc.WithTransportCredentials(creds))
+	require.NoError(t, err)
+
+	// Send a request for statistics with no data.
+	Eventually(func() error {
+		res, err := statsClient.List(ctx, &proto.StatisticsRequest{})
+		if err != nil {
+			return err
+		}
+		if res != nil {
+			return fmt.Errorf("statistics returned")
+		}
+		return nil
+	}, 5*time.Second, 1*time.Second).Should(Succeed())
+
+	// Create some flow data.
+	// Create a client to pusher Flows.
+	pusher, err := client.NewFlowClient(goldmaneURL, clientCert, clientKey, clientCA)
+	require.NoError(t, err)
+
+	connected := pusher.Connect(ctx)
+	require.NoError(t, err)
```

Comment:
Looks like here we check `err` from the `client.NewFlowClient` function call but not from `pusher.Connect(ctx)`

Reply:
`pusher.Connect` doesn't return an error though?